src/providers/configs/base.rs

```diff
@@ -0,0 +1,22 @@
+use anyhow::Result;
+use std::env;
+
+pub trait ProviderConfig {
```

Comment:
nit: curious what's the idea behind the provider config being separate from the provider?

Reply:
i was thinking providers have their way to authenticate so that's what the config represents